src/active_query.rs

```diff
@@ -91,7 +91,7 @@ impl ActiveQuery {
     ) {
         self.durability = self.durability.min(durability);
         self.changed_at = self.changed_at.max(changed_at);
-        self.input_outputs.insert(QueryEdge::Input(input));
+        self.input_outputs.insert(QueryEdge::input(input));
```

Comment:
The other approach that we could have considered (I don't think order matters between inputs and outputs) is to store the inputs and outputs (which most queries don't have) in separate boxed slices (and outputs in a thinvec).

Reply:
That might also slightly improve iteration over inputs or outputs, but the extra allocation might be costly.